packages/mdx/lib/plugin/recma-jsx-build.js

```diff
@@ -45,13 +45,15 @@ export function recmaJsxBuild(options) {
 
     // Remove the pragma comment that we injected ourselves as it is no longer
     // needed.
-    if (
-      tree.comments &&
-      tree.comments[0].type === 'Block' &&
-      tree.comments[0].data &&
-      tree.comments[0].data._mdxIsPragmaComment
-    ) {
-      tree.comments.shift()
+    if (tree.comments) {
+      let index = 0
+      while (index < tree.comments.length) {
+        if (tree.comments[index].data?._mdxIsPragmaComment) {
+          tree.comments.splice(index, 1)
+        } else {
+          index++
+        }
+      }
```

Comment:
```suggestion
    if (tree.comments) {
      tree.comments = tree.comments.filter(function (d) {
        return !d.data || !d.data._mdxIsPragmaComment
      })
```

Simpler?

Reply:
This creates a copy instead of modifying the original array. Fine with me, but I want to make sure you’re aware.

Also even simpler:

```suggestion
    if (tree.comments) {
      tree.comments = tree.comments.filter(function (d) {
        return !d.data?._mdxIsPragmaComment
      })
```

Or even:

```suggestion
    if (tree.comments) {
      tree.comments = tree.comments.filter((d) => !d.data?._mdxIsPragmaComment)
```

Reply:
I prefer real functions instead of arrows. `?.` is fine by me.
Both real functions and optional are in the rest of the code base.

New array is fine. This isn‘t a slow path. I’m more worried about iterating and splicing combined, easy to get wrong and hard to get fast in my micromark experience